storage/Filesystems/BtrfsImpl.cc

```diff
@@ -88,6 +113,28 @@ namespace storage
     }
 
 
+    FilesystemUser*
+    Btrfs::Impl::add_device(BlkDevice* blk_device)
+    {
+	if (blk_device->num_children() != 0)
+	    ST_THROW(WrongNumberOfChildren(blk_device->num_children(), 0));
+
+	FilesystemUser* filesystem_user = FilesystemUser::create(get_devicegraph(), blk_device, get_non_impl());
+
+	return filesystem_user;
+    }
+
+
+    void
+    Btrfs::Impl::remove_device(BlkDevice* blk_device)
```

Comment:
No check if `blk_device` is actually part of this Btrfs?
The subsequent instructions in the function may fail in spectacular ways if a caller passes a wrong block device or even a null pointer.

Reply:
The find_holder will throw an exception if the block device is not part of the btrfs.

I will add checks for null.